### DISCUSSION TITLE
error thrown while shell script ececution.

### DISCUSSION THREAD

**Discussion by abhinav9898** [Guvi-MentorDE/BE_batch_Guvi_Aug6#51] [General]
prera@abhinav-cluster1-m:~/bin$ ls
daily_script.sh  scd.hql
prera@abhinav-cluster1-m:~/bin$ sh daily_script.sh 20230901
: not foundt.sh: 2:
: not foundt.sh: 5:
: not foundt.sh: 7:
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/lib/tez/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/lib/hadoop/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Reload4jLoggerFactory]
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/lib/tez/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/lib/hadoop/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Reload4jLoggerFactory]
Hive Session ID = eef864db-9dff-4095-8729-e0bea1d8ff99

Logging initialized using configuration in file:/etc/hive/conf.dist/hive-log4j2.properties Async: true
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.hadoop.hive.common.StringInternUtils (file:/usr/lib/hive/lib/hive-common-3.1.3.jar) to field java.net.URI.string
WARNING: Please consider reporting this to the maintainers of org.apache.hadoop.hive.common.StringInternUtils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Hive Session ID = 4b871a84-94e7-48be-b447-416d3cf9e912
OK
Time taken: 1.218 seconds
OK
Time taken: 0.089 seconds
OK
Time taken: 0.197 seconds
OK
Time taken: 0.387 seconds
.txt'': No files matching path file:/home/prera/data/emp_data_20230901%0D.txt20230901

**Comment by kannankandasamy** [comment]
I had similar issues, for me it had some extra spaces in daily_script.sh, what i did was just opened that .sh file in vi editor and removed all extra spaces, and save it and try running, it might help, Good luck

**Reply by abhinav9898** [reply]
Hi @kannankandasamy , thanks for your response.
I tried removing all possible spaces, but still facing below error.

'': No files matching path file:/home/prera/data/emp_data_20230901%0D.txt%0D_20230901
  
Below is screenshot of the file in vi editor, I have given full permissions(777) as well.
![image](https://github.com/Guvi-MentorDE/BE_batch_Guvi_Aug6/assets/49013722/9322bbf0-7fbc-4915-9a6c-1b8776d1cd13)

Actual raw data File also exists in local path,
![image](https://github.com/Guvi-MentorDE/BE_batch_Guvi_Aug6/assets/49013722/1b277cb3-9426-4379-9abe-e4b3070acbb3)

**Reply by Guvi-MentorDE** [reply]
@abhinav9898  I would recomment dont copy the shell script. There are only 4 lines of code in the shell script. Please type it out by using Vi <script.sh> followed by chmod +x <script.sh> then try executing. We shall discuss alternate approaches to resolve spacing issues in next class.

**Reply by kannankandasamy** [reply]
Just to troubleshoot whether the issue is with .sh script you can try running hive directly as below:

hive --hiveconf file_path="/home/prera/data/emp_data_20230901.txt" -f /home/prera/bin/scd.hql

if this executes successfully then your first file executed, which means it is problem with script file only not on hive, try checking that

**Reply by abhinav9898** [reply]
Thanks @Guvi-MentorDE ,
It's resolved now, still not sure where it was taking the extra space. Will get it clarified next class.
